File: xigglyxuff.py
#!/usr/bin/python
import random, sys, time

# ints = [1,2,3,4,5,6,7,8,9]
# squares = [i*i for i in ints if i%2 == 0 else 0]

listy = ['b', 'd', 'f', 'g', 'j', 'l', 'm', 'n',
          'r', 's', 't', 'w', 'z', 'br', 'bl', 'p',
          'dr', 'fl', 'fr', 'gl', 'gr', 'pl', 'pr',
          'sn', 'sp', 'st', 'sw', 'tr', 'tw', 'wr',
          'ch', 'cl', 'kl', 'kr', 'schw', 'sch',
          'schl', 'y']

e1 = 'iggly'
e2 = 'uff'

print('▀▄▒▄▀ ▀█▀ ▒█▀▀█ ▒█▀▀█ ▒█░░░ ▒█░░▒█ ▀▄▒▄▀ ▒█░▒█ ▒█▀▀▀ ▒█▀▀▀ █ ')
print('░▒█░░ ▒█░ ▒█░▄▄ ▒█░▄▄ ▒█░░░ ▒█▄▄▄█ ░▒█░░ ▒█░▒█ ▒█▀▀▀ ▒█▀▀▀ ▀ ')
print('▄▀▒▀▄ ▄█▄ ▒█▄▄█ ▒█▄▄█ ▒█▄▄█ ░░▒█░░ ▄▀▒▀▄ ░▀▄▄▀ ▒█░░░ ▒█░░░ ▄ ')
print('\nHello, pokédude!')
print('\nWhat language do you speak, human? (Russian/ru, English/en, Robotic/ro, Potato/po)')
l1 = str(input())

if l1 == 'en':
    print('English selected.')
elif l1 == 'ru':
    listy = ['в', 'd', 'f', 'g', 'j', 'l', 'м', 'п',
            'ґ', '₴', 'т', 'ш', 'z', 'Ьґ', 'вl', 'p',
            'dґ', 'fl', 'fґ', 'gl', 'gя', 'pl', 'pя',
            '₴п', '$p', '₴т', '₴щ', 'тґ', 'тш', 'щя',
            'cн', 'cl', 'кl', 'кя', '$cнш', '$cн',
            '$cнl', 'у']
    e1 = 'їgglу'
    e2 = 'цff'
    print('Яц₴$їaп ₴ёlёcтёd.')
elif l1 == 'ro':
    listy = ['┬BZZZZZT', '╒BZZZZT', '║first law of robotics: A robot may not injure a human being or, through inaction, allow a human being to come to harm.', '╔', '∩01100', '®', '©', '¥',
            '¢TARGET ACQUIRED. ENGAGING.', 'ONEZEROZEROONEONEZEROONEONEZEROONEZEROZEROONEONEƒ', '$Captain, I have detected a seismic disturbance on the planet\' surface. It seems to be emanating fro-BKJAOERRORIJDERRORUOSERRORAIJERRORFLIAUERRORDSHFOERRORLIUr', '×', '™', '³', '²', 'Δ',
            '0110100101001101100010111010110010011001', '❤third law of robotics: A robot must protect its own existence as long as such protection does not conflict with the First or Second Law.', 'BOOP❥', '웃', '유', '♋', '☮', '✌',
            '☏DANGER-HOSTILES-DETECTED', '☢EXTERMINATE', '☠KILL ALL HUMANS!', '♚DESTROY. DESTROY.', '100101001100011011011001001101010100', '✔?', '♪', '✈',
            '⌘404 DOES NOT COMPUTE', '♂BEEP', '♀!second law of robotics: A robot must obey the orders given to it by human beings, except where such orders would conflict with the First Law.', '❅', 'ツ', '00101001☁', '☃', '✄',
            '☣if i only had a heeeaarrrtt', '☯RESISTANCE IS FUTILE']
    e1 = '♒✎∞☤✪'
    e2 = '♛♫✫'
    print('✉ღ✘BEEP✍✯☭➳❝ ✡✿BOOP℉❣')
#elif l1 == 'po':
#    listy = []
#    e1 = ''
#    e2 = ''
else:
    print('"'+l1+'" is an invalid option. Defaulting to English.')

while True:
    try:
        print('\nHow many usernames do you want me to generate? (q to quit)')
        u1 = input()
        
        if u1 == 'q':
            sys.exit(0)
            
        start = time.time()
        u1 = int(u1)
                    
        xigList = ['Wigglytuff', 'Jigglypuff', 'Igglybuff']

        if u1 < 1 or u1 > 500:
            print('\nWOAH! Are you crazy? That number is totally ridiculous.')
            continue
            
        for i in range(u1):
            xig = ''
            c1 = random.choice(listy)
            c2 = random.choice(listy)
            while c2 == c1:
                c2 = random.choice(listy)
            xig = c1.title()+e1+c2+e2
            if i % 4:
                print('\t', xig, end="")
            else:
                print('\n\t', xig, end="")
            xigList.append(xig)

        end = time.time()
        elapsed = end - start
        
        print('\n\nTime taken:', elapsed, 'seconds.')
        print('\nProfessor Oak thinks', random.choice(xigList), 'is a good choice.')
        del xigList[:]
          
    except ValueError:
        print('\nOnly integers are accepted. Sort your life out.')

File: xigglyxuff.pl
#!/usr/bin/perl
use strict;
use warnings;

print "Jello, Whirled!\n";
my ($c1, $c2, $c3);

my @xigList = qw(b d f g j l m n r s t w z br bl p dr fl fr gl gr pl pr sn sp st sw tr tw wr ch cl kl kr sch y);

print "\nHow many rows do you want?\n";
chomp(my $rows = <STDIN>);

if ($rows eq "\n" or $rows =~ /\D/) {
	print "Invalid input!";
	<>;
	exit;
} else {
	print "You chose: " . $rows . " rows.\n";
}

print "How many columns do you want? (must be an integer less than 7)\n";
chomp(my $columns = <STDIN>);

if ($columns eq "\n" or $columns =~ /\D/ or $columns > 6) {
	print "Invalid input!";
	<>;
	exit;
} else {
	print "You chose: " . $columns . " columns.\n";
}

my $total_numbers = $columns * $rows;

if ($total_numbers > 9999) {
	print "Too many numbers!";
	<>;
	exit;
}

my @num_array = ();
my ($box, $d, $f, $g, $h, $j);
my $line_segment = ("+-" . ("-" x 13) . "-");
my $a = 13 + 3;
my $line =  $line_segment x $columns;
my $linelength = length($line) * $columns;

if (length($linelength) > $columns) {
	print "Too many numbers!";
	<>;
	exit;
}

push(@num_array, " $line+\n");

print "\nHere are the usernames you requested.\n\n";

for (my $i = 1; $i <= $total_numbers; $i++) {
	$c1 = $xigList[rand @xigList];
	$c2 = $xigList[rand @xigList];
	while ($c2 eq $c1) {
		$c2 = $xigList[rand @xigList];
	}
	$j = ucfirst($c1) . "iggly" . $c2 . "uff";
	$g = ($a - length($j) - 1) / 2;
	$h = ($a - length($j)) / 2;
	$d = " " x $g;
	$f = " " x ($h - 1);
	$box = " |" . $d . $j . $f;
	
	if ($i%$columns == 0) {
			push(@num_array, $box . " |\n $line+\n");
		} else  {
			push(@num_array, $box);
		}
}

print @num_array;

<>;